be/src/exec/pipeline/exchange/sink_buffer.cpp

```diff
@@ -0,0 +1,215 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exec/pipeline/exchange/sink_buffer.h"
+
+namespace starrocks::pipeline {
+
+SinkBuffer::SinkBuffer(RuntimeState* state, const std::vector<TPlanFragmentDestination>& destinations,
+                       size_t num_sinkers)
+        : _mem_tracker(state->instance_mem_tracker()),
+          _brpc_timeout_ms(std::min(3600, state->query_options().query_timeout) * 1000),
+          _num_uncancelled_sinkers(num_sinkers) {
+    for (const auto& dest : destinations) {
+        const auto& instance_id = dest.fragment_instance_id;
+
+        auto it = _num_sinkers.find(instance_id.lo);
+        if (it != _num_sinkers.end()) {
+            it->second += num_sinkers;
+        } else {
+            _num_sinkers[instance_id.lo] = num_sinkers;
+
+            _request_sequences[instance_id.lo] = 0;
+            _max_processed_sequences[instance_id.lo] = -1;
+            _unprocessed_sequences[instance_id.lo] = std::unordered_set<int64_t>();
+            _buffers[instance_id.lo] = std::queue<TransmitChunkInfo, std::list<TransmitChunkInfo>>();
+            _num_finished_rpcs[instance_id.lo] = 0;
+            _num_in_flight_rpcs[instance_id.lo] = 0;
+            _mutexes[instance_id.lo] = std::make_unique<std::mutex>();
+
+            PUniqueId finst_id;
+            finst_id.set_hi(instance_id.hi);
+            finst_id.set_lo(instance_id.lo);
+            _instance_id2finst_id[instance_id.lo] = std::move(finst_id);
+        }
+
+        _num_remaining_eos = 0;
+        for (auto& [_, num] : _num_sinkers) {
+            _num_remaining_eos += num;
+        }
+    }
+}
+
+SinkBuffer::~SinkBuffer() {
+    DCHECK(is_finished());
+
+    for (auto& [_, buffer] : _buffers) {
+        while (!buffer.empty()) {
+            auto& request = buffer.front();
+            request.params->release_finst_id();
+            buffer.pop();
+        }
+    }
+}
+
+void SinkBuffer::add_request(const TransmitChunkInfo& request) {
+    DCHECK(_num_remaining_eos > 0);
+    if (_is_finishing) {
+        request.params->release_finst_id();
```

Comment:
`request.params->release_finst_id();` here is weired,  add comment to explain why and prevent someone from removing this statement by mistake.